### PR TITLE
melonDS: added recipe

### PR DESCRIPTION
**Pull request by Luca1991**
Hi all,
this is a recipe for melonDS (standalone).

I had a problem with devel:libX11.... I don't know why but this only works using "libX11${secondaryArchSuffix}_devel".
If you know what's the issue here, please tell me, so I can fix it.

Please keep the two patches separate: 
- "enable_wayland_flag" was made ny another dev and is already merged in upstream mainline.
- "melonds-0.9.5" is made by me and will be upstreamed after this merge.

Thanks a lot,
Luca

### CODE REVIEW

File: games-emulation/melonds/melonds-0.9.5.recipe
Comment (Begasus) [inline games-emulation/melonds/melonds-0.9.5.recipe:43]: In this case you might want to use `xlib${secondaryArchSuffix}_devel` instead of `libX11${secondaryArchSuffix}_devel` (not checked).

Comment (Luca1991) [reply games-emulation/melonds/melonds-0.9.5.recipe:43]: Do you mean xlibe?

Comment (Begasus) [reply games-emulation/melonds/melonds-0.9.5.recipe:43]: My bad :) xlibe${seconddaryArchSuffix}_devel indeed.

File: games-emulation/melonds/patches/enable_wayland_flag.patch
Comment (Begasus) [inline games-emulation/melonds/patches/enable_wayland_flag.patch:1]: Could you created a patchset here, it can consist of multiple commits also, running with -e after extract will combine the 2 patches into one.

Comment (Luca1991) [reply games-emulation/melonds/patches/enable_wayland_flag.patch:1]: Yes I can, but I'm not the author of the "enable_wayland_flag" patch (it was made by an upstream contributor). If a patchset is preferred, I can create one (using the usual commands) and then give proper credits adding a comment in the final file. Please let me know.

Comment (Begasus) [reply games-emulation/melonds/patches/enable_wayland_flag.patch:1]: You can grab that one with `wget https://github.com/melonDS-emu/melonDS/pull/1654.patch` (that includes the auriginal author) :)

File: games-emulation/melonds/patches/melonds-0.9.5.patch
Comment (Begasus) [inline games-emulation/melonds/patches/melonds-0.9.5.patch:1]: This one could be a good candidate for upstream +1

File: games-emulation/melonds/melonds-0.9.5.recipe
Comment (Begasus) [inline games-emulation/melonds/melonds-0.9.5.recipe:19]: No capitols here:
```diff
-	melonDS$secondaryArchSuffix = $portVersion
+	melonds$secondaryArchSuffix = $portVersion
```

File: games-emulation/melonds/melonds-0.9.5.recipe
Comment (Begasus) [reply games-emulation/melonds/melonds-0.9.5.recipe:43]: Builds fine with xlibe :)

File: games-emulation/melonds/melonds-0.9.5.recipe
Comment (Begasus) [inline games-emulation/melonds/melonds-0.9.5.recipe:41]: libSDL* (also in REQUIRES)
```diff
-	devel:libsdl2_2.0$secondaryArchSuffix
+	devel:libSDL2_2.0$secondaryArchSuffix
```

File: games-emulation/melonds/patches/melonds-0.9.5.patch
Comment (Luca1991) [reply games-emulation/melonds/patches/melonds-0.9.5.patch:1]: Yes, this is the patch that I'll upstream after the merge of this PR :)

**Review by Begasus — CHANGES_REQUESTED** [review]

File: games-emulation/melonds/melonds-0.9.5.recipe
Comment (Begasus) [inline games-emulation/melonds/melonds-0.9.5.recipe:11]: Use $portVersion here:
```diff
-PATCHES="melonds-0.9.5.patchset"
+PATCHES="melonds-$portVersion.patchset"
```

File: games-emulation/melonds/melonds-0.9.5.recipe
Comment (Luca1991) [reply games-emulation/melonds/melonds-0.9.5.recipe:5]: Ok no problem, I'll remove it soon. But I think that it is good to have a package submitter/maintainer (like in AUR), please take this idea in consideration for the future :)

**Comment by Luca1991** [comment]
I've just pushed the proposed changes :)

File: games-emulation/melonds/melonds-0.9.5.recipe
Comment (Begasus) [reply games-emulation/melonds/melonds-0.9.5.recipe]: These are stored in githubs logs for now, don't have a strong opinion here, but so far no one ever used it. :)

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
Thanks! 👍

**Comment by Begasus** [comment]
Build error on 32bit: https://build.haiku-os.org/buildmaster/master/x86_gcc2/logviewer.html?buildruns/5259/builds/82151.log

**Comment by Luca1991** [comment]
@Begasus doesn't "!x86_gcc2" prevents the building of the 32bit package?

**Comment by Begasus** [comment]
It prevents it from building with gcc2 (not 32bit), with `SECONDARY_ARCHITECTURES="x86"` it can/could be build on 32bit with gcc13.

**Comment by Luca1991** [comment]
@Begasus oh I understand. Since I can't test on 32bit, can I just disable the building on anything that isn't x64?

**Comment by Begasus** [comment]
You can, but it would be a shame if you do, it would be better if we could fix this first.

**Pull request merged**
